edgelet/edgelet-docker/src/notary.rs

```diff
@@ -94,6 +97,39 @@ pub fn notary_init(homedir: &Path, hostname: &str, path: &PathBuf) -> Result<Pat
     Ok(config_file_path)
 }
 
+pub fn notary_lookup(
+    notary_auth: &str,
+    image_gun: &str,
+    tag: &str,
+    config_path: &str,
+    lock: tokio::sync::lock::LockGuard<()>,
+) -> impl Future<Item = String, Error = Error> {
+    Command::new("notary")
+        .env("NOTARY_AUTH", notary_auth)
+        .arg("lookup")
+        .args(&[image_gun, tag])
+        .args(&["-c", config_path])
+        .output_async()
+        .then(|notary_output| {
+            drop(lock);
+            let notary_output = notary_output.with_context(|_| {
+                ErrorKind::LaunchNotary("could not spawn notary process".to_owned())
+            })?;
+            let notary_output_string =
+                String::from_utf8(notary_output.stdout).with_context(|_| {
+                    ErrorKind::LaunchNotary("could not retrieve notary output as string".to_owned())
+                })?;
+            let split_array: Vec<&str> = notary_output_string.split_whitespace().collect();
+            if split_array.len() < 2 {
+                return Err(ErrorKind::LaunchNotary(
+                    "notary digest split array is empty".to_owned(),
```

Comment:
From customers' perspective, what can be the reason for this to be hit?

Reply:
The output from notary lookup results in a format like this : **tag digest bytes** example : **v1 sha256:a15790640a6690aa1730c38cf0a440e2aa44aaca9b0e8931a9f2b0d7cc90fd65 528**. this is the resultant array. The goal is get this split array. Very rarely the customer can hit this error with this error being empty. Any other major issues will be hit before line 125.